databroker/src/broker.rs

```diff
@@ -1561,6 +1623,290 @@ impl<'a, 'b> AuthorizedAccess<'a, 'b> {
             Err(e) => Err(QueryError::CompilationError(format!("{e:?}"))),
         }
     }
+
+    pub async fn provide_actuation(
+        &self,
+        vss_ids: Vec<i32>,
+        actuation_provider: Box<dyn ActuationProvider + Send + Sync + 'static>,
+    ) -> Result<(), (ActuationError, String)> {
+        for vss_id in vss_ids.clone() {
+            let result_entry = self.get_entry_by_id(vss_id).await;
+            match result_entry {
+                Ok(entry) => {
+                    let vss_path = entry.metadata.path;
+                    let result_can_write_actuator =
+                        self.permissions.can_write_actuator_target(&vss_path);
+                    match result_can_write_actuator {
+                        Ok(_) => {}
+                        Err(PermissionError::Denied) => {
+                            let message = format!("Permission denied for vss_path {}", vss_path);
+                            return Err((ActuationError::PermissionDenied, message));
+                        }
+                        Err(PermissionError::Expired) => {
+                            return Err((
+                                ActuationError::PermissionExpired,
+                                "Permission expired".to_string(),
+                            ))
+                        }
+                    }
+                }
+                Err(ReadError::NotFound) => {
+                    let message = format!("Could not resolve vss_path of vss_id {}", vss_id);
+                    return Err((ActuationError::NotFound, message));
+                }
+                Err(ReadError::PermissionDenied) => {
+                    let message = format!("Permission denied for vss_id {}", vss_id);
+                    return Err((ActuationError::PermissionDenied, message));
+                }
+                Err(ReadError::PermissionExpired) => {
+                    return Err((
+                        ActuationError::PermissionExpired,
+                        "Permission expired".to_string(),
+                    ))
+                }
+            }
+        }
+
+        let provided_vss_ids: Vec<i32> = self
+            .broker
+            .subscriptions
+            .read()
+            .await
+            .actuation_subscriptions
+            .iter()
+            .flat_map(|subscription| subscription.vss_ids.clone())
+            .collect();
+        let intersection: Vec<_> = vss_ids
```

Comment:
```suggestion
        let intersection: Vec<i32> = vss_ids
```
rust can identify that Vec<_> should be Vec<i32> but shouldn't we just specify that it is that? I find it more readable this way. Whats your opinion @argerus @rafaeling